src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
@mavasani issue is compiler analyzer's SupportedDiagnostics is wrong. since it reports diagnostics that it can never produce as supported diagnostics.